pkg/cmd/image/list.go

```diff
@@ -188,50 +189,126 @@ type imagePrinter struct {
 	quiet, noTrunc, digestsFlag, namesFlag bool
 	tmpl                                   *template.Template
 	client                                 *containerd.Client
-	contentStore                           content.Store
+	contentStore                           containerdutil.Provider
 	snapshotter                            snapshots.Snapshotter
 }
 
-func (x *imagePrinter) printImage(ctx context.Context, img images.Image) error {
-	ociPlatforms, err := images.Platforms(ctx, x.contentStore, img.Target)
+type Image struct {
```

Comment:
This and following is the biggest part of the refactoring of `images`.
As mentioned earlier, what we currently have is making many unnecessary calls / iterations.
This is closer to what an image is on the store, and minimizes reads and roundtrips to the store.

This Image struct here is ad-hoc - but we might consider making these methods into utils and revamping some of our abstractions later on.